x/crosschain/keeper/msg_server_vote_inbound_tx.go

```diff
@@ -192,16 +193,17 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 
 				// gas payment for erc20 type might fail because no liquidity pool is defined to swap the zrc20 token into the gas token
 				// in this gas we should refund the sender on ZetaChain
-				if cctx.InboundTxParams.CoinType == common.CoinType_ERC20 {
-
-					if err := k.RefundAmountOnZetaChain(ctx, cctx, cctx.InboundTxParams.Amount); err != nil {
+				if cctx.InboundTxParams.CoinType == common.CoinType_ERC20 && ethcommon.IsHexAddress(cctx.InboundTxParams.Sender) {
+					// Sender is verified to be a valid ethereum address
+					if err := k.RefundAbortedAmountOnZetaChain(ctx, cctx, ethcommon.HexToAddress(cctx.InboundTxParams.Sender)); err != nil {
 						// log the error
 						k.Logger(ctx).Error("failed to refund amount of aborted cctx on ZetaChain",
 							"error", err,
 							"sender", cctx.InboundTxParams.Sender,
 							"amount", cctx.InboundTxParams.Amount.String(),
 						)
 					}
+					cctx.CctxStatus.IsAbortRefunded = true
```

Comment:
should we update the logic as below?
```
else {
    cctx.CctxStatus.IsAbortRefunded = true
}
```

Reply:
792f4c99e4a2cb459892a35b74996f382a737cb7